content/blog/eng/automation-testing-a-step-back.md

```diff
@@ -0,0 +1,114 @@
+Computers are great at scaling and automation! So developers at companies large and small write automated tests, which are tests that are run by software and do not require you to manually operate the app to verify the code works.
+
+This is the introduction of a Google test codelab. It not only defines in a pretty straightforward way what automated tests are, but it also tells why you should do that. The reason mentioned here is only the first of the four on the list that I have compiled before, but this is the main reason.
+
+So we write automated tests to save time in the future, it's an investment. We suppose that when we add more features the old ones need to be checked again, and we invest our time in order to save more in the future, but in the short term we will not have a return.
+
+If we imagine a big project with many developers it becomes more clear, we cannot check manually everything every time (if we want to release often or with simplicity). So if we want to release soon and we have no automation tests we end up testing only what seems most important at the moment, exposing ourselves to the risk of regression.
+
+## Count as documentation
+
+This is an easy point. Documentation can be important (less than working software, see agile manifesto) but it's a pain for most and takes a lot of time to do it well. But if tests are written in a certain way they can have the same information of product documentation, that evolves and adapts as the software grows. This way the software can be self-documented.
+
+## Help write better code
+
+This is a very generic sentence, what do I mean by "better code"? First say that software as every other built thing has a quality. A program can be better than another that does the same things based on some parameters (performance, usability and many others), we can divide these parameters into 2 big groups: the one that users notice and the one that users don't.
+
+For example, if the same thing code is written without naming convection or indentation the user doesn't notice and the software runs exactly the same, so why should we care? Because if we want to modify this code, adding new features for example, it will be much much harder.
+
+That said a code is "better" than another when, even if does the same things, it is easier to maintain and to evolve (beeing more readable, respecting the SOLID principles, etc...).
+
+Test helps to increase quality because to make a portion of code testable we are often forced to restructure it and change the management of dependencies. This generally translates into a more correct distribution of responsibilities. So it's better from an architectural point of view.
```

Comment:
Here maybe we can distinguish between test automation and tdd perhaps?

Reply:
Test automation can be done with or without tdd. I want to keep the article as simple as possible without using concepts like tdd, which in my opinion come one step later

Reply:
I totally agree with you. In fact, maybe be useful to show the difference. 
Many times I saw to confuse TDD with test automation. For me TDD is a design technique, test automation is a tool.
But if for you isn't correct ok!